### PR TITLE
Feat: CoreParagraph Block

### PR DESCRIPTION
**Pull request by theodesp**
## Tasks

- [ ] I have signed a [Contributor License Agreement (CLA)](https://github.com/wpengine/faustjs#contributor-license-agreement) with WP Engine.

## Description

* Provides reference implementation of CoreParagraph block. 
* Uses `getStyles` helper/
* Adds Unit tests.

## Testing

* Follow instructions to configure the `blocks` package in the example project. https://faustjs.org/docs/gutenberg/getting-started
* Generate the global stylessheet and import it into the project. `faust generateGlobalStylesheet`.
```_app.js
import { WordPressBlocksProvider, fromThemeJson } from '@faustwp/blocks';
import themeJson from '../theme.json';
import '../globalStylesheet.css';

export default function MyApp({ Component, pageProps }) {
  const router = useRouter();
  return (
    <FaustProvider pageProps={pageProps}>
      <WordPressBlocksProvider
        config={{
          blocks,
          theme: fromThemeJson(themeJson),
        }}>
        <Component {...pageProps} key={router.asPath} />
      </WordPressBlocksProvider>
    </FaustProvider>
  );
}
```
* Comment out all the styles in the `examples/next/faustwp-getting-started/styles/_blocks.scss` since they will interfere with the global stylesheet.
* Import the blocks:
```js
import { coreBlocks } from '@faustwp/blocks';

export default {
  ...coreBlocks,
};
```
* Make sure you use the block fragment in the page query:

```graphql
${components.CoreParagraph.fragments.entry}
...
editorBlocks {
        __typename
        renderedHtml
        id: clientId
        parentClientId
        ...${components.CoreParagraph.fragments.key}
      }
```
* Use the `WordPressBlocksViewer`:
```
<Container>
            <ContentWrapper>
              <WordPressBlocksViewer blocks={blocks}/>
            </ContentWrapper>
          </Container>
```
* Go to WordPress site and create several paragraphs combining styles and colors. All changes should reflect in the decoupled site when updating the post.

## Screenshots

Screenshot 2023-05-24 at 12 05 57

### CODE REVIEW

**Review by josephfusco — COMMENTED** [review]
Everything looks good except for the padding and link color. Curious if others can reproduce this or if I'm missing a piece.

Screenshot 2023-05-26 at 10 16 27 AM

**Comment by theodesp** [comment]
@josephfusco you need to remove some of the `_base.scss` styles for body and paragraph as they interfere with the global styles:

```
// Base styles

// *,
// *::before,
// *::after {
// 	box-sizing: border-box;
// }

// p {
// 	margin: 0 0 1rem;
// 	line-height: 1.8;
// }
```

Also you may need to add some additional classes for container and contentWrapper:

```
<Main className="wp-site-blocks">
        <>
          <EntryHeader
            title={title}
            image={featuredImage?.node}
            date={date}
            author={author?.node?.name}
          />
          <Container className="wp-block-group is-layout-flow">
            <ContentWrapper className="entry-content wp-block-post-content has-global-padding is-layout-constrained">
              <WordPressBlocksViewer blocks={blocks}/>
            </ContentWrapper>
          </Container>
        </>
      </Main>
```

**Comment by mindctrl** [comment]
@theodesp I see the same as @josephfusco ... padding and link colors do not look like the editor side. I haven't tried the fix you suggested to Joe, mainly because I'm curious what the plan is here to address it? We can't expect users to catch a comment on this PR to fix the issues with padding and colors, so is there a plan to fix this in another PR, update the example, or similar?

**Comment by theodesp** [comment]
Hey @mindctrl ideally you may want with to use a blank theme and you want to comment the styles in `_base.scss`. This is what I can see when I do that:

**Editor View**
Screenshot 2023-05-31 at 15 43 56

**Decoupled View**
Screenshot 2023-05-31 at 15 44 00

If you have additional styles you may find that they overlap with the editor view. I will add instructions to the ticket.

**Comment by theodesp** [comment]
In regards to the link styles it mainly depends on the `theme.json` link control support. 

Gutenberg adds a lot of styles on the theme that are hard. to pinpoint exactly. For example here it adds some inline styles under the name: `core-block-supports-inline-css`:

Screenshot 2023-05-31 at 16 28 01

These are difficult to control on the theme side. According to the docs you will have to use the following filters:

https://fullsiteediting.com/lessons/how-to-remove-default-block-styles/#h-how-to-remove-the-inline-styles-on-the-front

```
remove_filter( 'render_block', 'wp_render_elements_support', 10, 2 );
remove_filter( 'render_block', 'gutenberg_render_elements_support', 10, 2 );
```

in other words ideally we want to start with a blank theme.json and a blank theme in headless site to make styles as close as possible.

**Review by mindctrl — APPROVED** [review]
